extensions/jira/deployment/src/main/java/org/apache/camel/quarkus/component/jira/deployment/JiraProcessor.java

```diff
@@ -80,4 +87,38 @@ List<ReflectiveClassBuildItem> registerReflectiveClasses() {
         items.add(ReflectiveClassBuildItem.builder("org.codehaus.jettison.json.JSONObject").methods(true).build());
         return items;
     }
+
+    @BuildStep(onlyIf = IsSpringBeansAbsent.class)
+    void generateDisposableInterface(BuildProducer<GeneratedClassBuildItem> generatedClass) {
+        // TODO: remove if https://ecosystem.atlassian.net/browse/JRJC-258 eventually gets fixed
+        try (ClassCreator classCreator = ClassCreator.interfaceBuilder()
+                .className(DISPOSABLE_BEAN_CLASS_NAME)
+                .classOutput(new GeneratedClassGizmoAdaptor(generatedClass, false))
+                .build()) {
+
+            /*
+             * Original implementation of DisposableBean is:
+             *
+             * public interface DisposableBean {
+             *   void destroy() throws Exception;
+             * }
+             */
+            try (MethodCreator methodCreator = classCreator.getMethodCreator("destroy", void.class)) {
+                methodCreator.setModifiers(Modifier.PUBLIC | Modifier.ABSTRACT);
+                methodCreator.addException(Exception.class);
+            }
+        }
+    }
+
+    static final class IsSpringBeansAbsent implements BooleanSupplier {
+        @Override
+        public boolean getAsBoolean() {
+            try {
+                Class.forName(DISPOSABLE_BEAN_CLASS_NAME);
```

Comment:
Is it better to load class with TCCL?
```suggestion
                Class.forName(DISPOSABLE_BEAN_CLASS_NAME, true, 
 Thread.currentThread().getContextClassLoader());
```

Reply:
Yes. Let me do that :+1:

Reply:
Done